components/Map/Map.jsx

```diff
@@ -0,0 +1,170 @@
+`;
+
+const style = {
+  width: "100%",
+  height: "50vh",
+};
+
+class Map extends Component {
+  componentDidMount() {
+    Promise.all([
+      fetch(
+        "https://docs.google.com/spreadsheets/d/e/2PACX-1vQ4l7bRcBIgwsEPGM_s9zF9csIeTgE2No_4tA6MuDCBUbfmWY_e9mAfzPpCJTsIK_hUzOyJ8CmdGMsX/pub?gid=641305740&single=true&output=csv"
+      )
+        .then((response) => response.text())
+        .then((csv) => {
+          const parsed = Papa.parse(csv, { header: true, dynamicTyping: true });
+          return Promise.resolve(parsed.data);
+        }),
+      fetch(
+        "https://raw.githubusercontent.com/bhrutledge/ma-legislature/main/dist/ma_house.geojson"
+      ).then((response) => response.json()),
+    ]).then(([supporters, houseFeatures]) => {
+      const supportersByDistrict = supporters.reduce((acc, cur) => {
+        acc[cur.district] = cur;
+        return acc;
+      }, {});
+
+      function repProperties(feature) {
+        const supporter = supportersByDistrict[feature.properties.district];
+        return supporter;
+      }
+
+      const houseLayer = L.geoJson(houseFeatures, {
+        style: (feature) => districtStyle(repProperties(feature)),
+        onEachFeature: (feature, layer) => {
+          const rep = repProperties(feature);
+          layer.bindPopup(districtPopup(rep));
+          layer.on("popupopen", onPopup);
+          layer.on("popupclose", onPopup);
+          feature.properties.index = `${rep.first_name} ${rep.last_name} - ${rep.district}`;
+        },
+      });
+
+      const searchControl = new L.Control.Search({
+        layer: houseLayer,
+        propertyName: "index",
+        initial: false,
+        marker: false,
+        textPlaceholder: "Search reps and districts",
+        moveToLocation(latlng, title, map) {
+          map.fitBounds(latlng.layer.getBounds());
+          latlng.layer.openPopup();
+        },
+      });
+
+      const legendControl = L.control({ position: "topright" });
+      legendControl.onAdd = () => {
+        const div = L.DomUtil.create("div", "legend");
+        div.innerHTML = districtLegend();
+        return div;
+      };
+      const baseLayer = L.tileLayer(
+        "http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png",
+        {
+          attribution:
+            '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors, &copy; <a href="http://cartodb.com/attributions">CartoDB</a>',
+        }
+      );
+
+      this.map = L.map("map")
+        .addLayer(baseLayer)
+        .addLayer(houseLayer)
+        .addControl(searchControl)
+        .addControl(legendControl)
+        .fitBounds(houseLayer.getBounds());
+
+      // Avoid accidental excessive zoom out
+      this.map.setMinZoom(this.map.getZoom());
+    });
+  }
+
+  render() {
+    return (
+      <Fragment>
+        <Head>
+          <link
+            rel="stylesheet"
+            href="https://unpkg.com/leaflet-search@2.9.9/dist/leaflet-search.min.css"
```

Comment:
Don't know a ton about React. Does this cause you to reload the js/css every time the map gets updated?